### PR TITLE
crowbar: On restore, only restore roles that are for proposals and nodes

### PR DESCRIPTION
**Pull request by vuntz**
We do not want to restore roles such as provisioner-base, as the role
should be considered as code and not data, and it might be outdated in
the backup.

### CODE REVIEW

**Comment by vuntz** [comment]
So I believe we should also have similar code when creating the backup...

File: crowbar_framework/lib/crowbar/backup/restore.rb [hunk 1]
```diff
@@ -191,6 +191,11 @@ def restore_chef
               file = Pathname.new(file)
               # skip client "crowbar"
               next if type == :clients && file.basename.to_s =~ /^crowbar.json$/
+              # skip roles which are not crowbar roles (proposal + node roles)
+              # the regexp for node roles is not great, but we know we have a _
+              # due to the domain, so it should be good enough
+              next if type == :roles &&
+                  file.basename.to_s !~ /.+-config-.+/ && file.basename.to_s !~ /^crowbar-.+_.+/
```
Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: I think you could put it into one regex
```ruby
file.basename.to_s !~ /(.+-config-.+)|(^crowbar-.+_.+)/
```

Comment (vuntz) [inline crowbar_framework/lib/crowbar/backup/restore.rb]: Good point, done

**Comment by vuntz** [comment]
Updated to filter on backup creation too.

File: crowbar_framework/lib/crowbar/backup/base.rb [hunk 2]
```diff
@@ -100,6 +100,29 @@ def restore_files_after_install
...
+
+        def filter_chef_client(name)
+          # Filter client "crowbar" as we need the new one on restore anyway
+          name =~ /^crowbar$/
+        end
+
+        def filter_chef_databag(db, name)
+          # Filter items in crowbar data bag that are not network databag items
+          return false if db != "crowbar"
+          item !~ /.+_network$/
```
Comment (MaximilianMeister) [inline crowbar_framework/lib/crowbar/backup/base.rb]: s/item/name

**Comment by MaximilianMeister** [comment]
codewise +1 (triggered an upgrade gating too)

**Comment by vuntz** [comment]
I added the patch from https://github.com/crowbar/crowbar-core/pull/562 to help gating.

**Comment by jsuchome** [comment]
+1 (tested)

**Comment by MaximilianMeister** [comment]
@vuntz once gating is unbroken and this is rebased, can you run another upgrade gating here?

**Comment by MaximilianMeister** [comment]
@vuntz I triggered the gating for the upgrade, I hope it works :) even though the last commit hash starts with "bad" :-(  https://github.com/crowbar/crowbar-core/pull/561/commits/bad8795df655872f37844167649b97be07324d09

**Comment by MaximilianMeister** [comment]
+1 we cannot gate the upgrade, because the admin gets reinstalled...

**Pull request merged**